src/Agent.Worker/Build/SourceProvider.cs

```diff
@@ -15,6 +16,10 @@ public interface ISourceProvider : IExtension, IAgentService
         string RepositoryType { get; }
 
         string GetBuildDirectoryHashKey(IExecutionContext executionContext, ServiceEndpoint endpoint);
+
+        Task GetSourceAsync(IExecutionContext executionContext, ServiceEndpoint endpoint, CancellationToken cancellationToken);
+
+        Task ResetRepositoryAsync(IExecutionContext executionContext, ServiceEndpoint endpoint);
```

Comment:
This name is misleading. I'm not sure of a better name offhand. `RemoveEmbeddedCredentialsAsync`? Or could go a generic route, e.g. something like `CleanupPostJobAsync`. I kind of prefer the non-generic name since that's it's only purpose *today*.